reflex-dom-core/src/Reflex/Dom/Builder/Class.hs

```diff
@@ -150,11 +150,41 @@ class (Monad m, Reflex t, DomSpace (DomBuilderSpace m), MonadAdjust t m) => DomB
                    => m ()
   notReady = lift notReady
 
+{- 2017-05-19 dridus: Commented out per Ryan Trinkle. Note that this does not support mount state and possibly has issues with unready child handling.
+
 class DomBuilder t m => MountableDomBuilder t m where
   type DomFragment m :: *
   buildDomFragment :: m a -> m (DomFragment m, a)
   mountDomFragment :: DomFragment m -> Event t (DomFragment m) -> m ()
 
+-}
+
+-- |'MonadMountStatus' represents an action that can be aware of whether the corresponding DOM built by the action is present within the document yet or not.
+-- Its primary use is to integrate with external libraries which need to be invoked only when DOM structures are installed in the document.
+--
+-- ___Note:___ once the current scope is replaced, any 'performEvent's in the scope will be cancelled and so if you want to observe the 'Unmounted' status
+-- you have to plumb the mount state dynamic back out of the current scope so the parent scope can react to it.
+class Monad m => MonadMountStatus t m | m -> t where
+  -- |Get a 'Dynamic' representing the current 'MountState' of DOM elements created in the current scope.
+  getMountStatus :: m (Dynamic t MountState)
+
+instance MonadMountStatus t m => MonadMountStatus t (ReaderT r m) where
+  getMountStatus = lift getMountStatus
+instance MonadMountStatus t m => MonadMountStatus t (PostBuildT t m) where
+  getMountStatus = lift getMountStatus
+
+-- |Type representing the current mount status of a DOM structure. Mount status refers to whether the DOM structure is currently within the document tree, not
+-- in the document tree, or transitioning.
+data MountState
+  -- note: order of these constructors is important, because Ord is derived and employed when combining parent and child mount states
+  = Unmounted
+  -- ^DOM structures have been removed from the document.
+  | Mounting
+  -- ^DOM structures are not yet installed in the document.
+  | Mounted
+  -- ^DOM structures are now in the document.
+  deriving (Eq, Ord, Show)
```

Comment:
I had it originally and removed it because I wasn't sure how to wire it in... I think it'd need to go to Unmounting when runWithReplace is about to remove it but before the DOM updates happen?